tests/integration/local/invoke/test_integrations_do_cli.py

```diff
@@ -0,0 +1,54 @@
+import os
+import locale
+import sys
+
+from io import StringIO
+from pathlib import Path
+
+from parameterized import parameterized
+import pytest
+
+from tests.integration.local.invoke.layer_utils import LayerUtils
+from .invoke_integ_base import InvokeIntegBase
+from tests.testing_utils import IS_WINDOWS, RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+
+from samcli.commands.local.invoke.cli import cli
+from samcli.cli.context import Context
+
+from click.testing import CliRunner
+
+
+class TestSamPython36HelloWorldNonUTF8Integration(InvokeIntegBase):
+    template = Path("template.yml")
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        def mock_getpreferredencoding(do_setlocale=True):
+            return "cp932"
+
+        cls._original_getpreferredencoding = locale.getpreferredencoding
+        locale.getpreferredencoding = mock_getpreferredencoding
+
+    @classmethod
+    def tearDownClass(cls):
+        locale.getpreferredencoding = cls._original_getpreferredencoding
+
+    @pytest.fixture(autouse=True)
+    def inject_fixtures(self, caplog):
```

Comment:
I am not super familiar with pytest fixtures. What is this doing and why should we use this over the unittest module in the standard lib? Historically, we have tried to keep all our test using soely unittest, so everything is consistent. I am ok with using fixtures for this, but in the future I would prefer to match our existing setup and not rely on pytest other than a runner of tests.

Reply:
caplog fixture is needed here for a issue with Click and Logging. 
https://github.com/pallets/click/issues/824
Not sure if there is another way around this problem.

Reply:
Works for me. Thanks for explaining.